action/action-dsl/src/main/java/com/powsybl/action/dsl/GroovyDslContingenciesProvider.java

```diff
@@ -45,18 +45,19 @@ public GroovyDslContingenciesProvider(final Path path) {
      */
     public  GroovyDslContingenciesProvider(final InputStream input) {
         Objects.requireNonNull(input);
-        this.dslLoader = createLoader(new InputStreamReader(input, StandardCharsets.UTF_8));
-    }
-
-    private static ActionDslLoader createLoader(final Reader reader) {
-        GroovyCodeSource src = new GroovyCodeSource(reader, "script", GroovyShell.DEFAULT_CODE_BASE);
-        return new ActionDslLoader(src);
+        this.script = new GroovyCodeSource(new InputStreamReader(input, StandardCharsets.UTF_8), "script", GroovyShell.DEFAULT_CODE_BASE);
     }
 
     @Override
     public List<Contingency> getContingencies(Network network) {
-        ActionDb actionDb = dslLoader.load(network);
+
+        ActionDb actionDb = new ActionDslLoader(script).load(network);
         return ImmutableList.copyOf(actionDb.getContingencies());
     }
 
+    @Override
+    public String asScript() {
```

Comment:
getScript?

Reply:
I thought "getScript" may let people think that this is just a simple getter,
while if we want later to implement a more generic translation to script (based on the getContingencies return for exemple), there will be some transformation underneath.

That's why I thought "asScript" or "toScript" may be better. Not very important though!

Note: in the case where we would generate the script from the list, it would need the Network as additional argument.